app/lib/actions/pulp/repository/sync.rb

```diff
@@ -27,11 +27,26 @@ def invoke_external_task
             sync_options[:validate] = !(SETTINGS[:katello][:pulp][:skip_checksum_validation])
 
             output[:pulp_tasks] = pulp_tasks =
-                pulp_resources.repository.sync(input[:pulp_id],  override_config: sync_options)
+                [pulp_resources.repository.sync(input[:pulp_id],  override_config: sync_options)]
+
             pulp_tasks
           end
         end
 
+        def external_task=(tasks)
+          output[:contents_changed] = content_changed?(tasks)
+          super
+        end
+
+        def content_changed?(tasks)
```

Comment:
Some places you use contents_changed and others you use content_changed. Might be nice to use the same terminology.